UnitySDK/Assets/ML-Agents/Editor/LearningBrainEditor.cs

```diff
@@ -13,15 +13,65 @@ namespace MLAgents
     [CustomEditor(typeof(LearningBrain))]
     public class LearningBrainEditor : BrainEditor
     {
+        private const string ModelPropName = "model";
+        private const string DevicePropName = "inferenceDevice";
+        private const float TimeBetweenModelReloads = 2f;
+        private float _timeSinceModelReload;
+        private bool _requireReload;
+        
+        /// <summary>
+        /// Is called once when the user opens the Inspector for the LearningBrain
+        /// </summary>
+        public void OnEnable()
+        {
+            _requireReload = true;
+            EditorApplication.update += Update;
+        }
+        /// <summary>
+        /// Is called once when the user leaves the Inspector for the LearningBrain
+        /// </summary>
+        public void OnDisable()
+        {
+            EditorApplication.update -= Update;
+        }
+        
         public override void OnInspectorGUI()
         {
             EditorGUILayout.LabelField("Learning Brain", EditorStyles.boldLabel);
             var brain = (LearningBrain) target;
             var serializedBrain = serializedObject;
+            EditorGUI.BeginChangeCheck();
             base.OnInspectorGUI();
             serializedBrain.Update(); 
-            EditorGUILayout.PropertyField(serializedBrain.FindProperty("graphModel"), true);
+            var tfGraphModel = serializedBrain.FindProperty(ModelPropName);
+            EditorGUILayout.ObjectField(tfGraphModel);
+            var deviceType = serializedBrain.FindProperty(DevicePropName);
+            EditorGUILayout.PropertyField(deviceType);
             serializedBrain.ApplyModifiedProperties();
+
+            if (EditorGUI.EndChangeCheck())
+            {
+                _requireReload = true;
+            }
+            if (_requireReload && _timeSinceModelReload > TimeBetweenModelReloads)
+            {
+                brain.SetModel(brain.model);
```

Comment:
We should have a brain.reloadModel() method that can reload the model instance it already has access too. The pattern `instance.set(instance.attribute)` is unintuitive.

Reply:
+1